### PR TITLE
[WIP] download archive instead of cloning repo in integration tests

### PR DESCRIPTION
**Pull request by lzutao**
changelog: none

### CODE REVIEW

**Comment by lzutao** [comment]
It works.

**Comment by lzutao** [comment]
Note that this PR will not lead to good win in build time. But it will not lead to
build regression either.

**Comment by phansch** [comment]
The integration tests took ~20 minutes instead of the usual 10 minutes this time. Let's try again; maybe a caching issue?

@bors try

**Comment by lzutao** [comment]
> maybe a caching issue?

Yes, it is. Because of #4692 , the cache in try branch is not updated.

**Comment by lzutao** [comment]
I will investigate the regression in build time later.

File: ci/integration-tests.sh [hunk 1]
```diff
@@ -10,8 +10,9 @@ cargo install --force --debug --path .
 
 echo "Running integration test for crate ${INTEGRATION}"
 
-git clone --depth=1 "https://github.com/${INTEGRATION}.git" checkout
```
Comment (flip1995) [inline ci/integration-tests.sh]: Couldn't this also be done with `--single-branch`?

Comment (lzutao) [reply ci/integration-tests.sh]: Yes. But downloading `tar.gz` archives is about 1 second faster than using git.

Comment (flip1995) [reply ci/integration-tests.sh]: So 1 second in a ~8 minute build? I think just using `git` should be ok then?

Comment (matthiaskrgr) [reply ci/integration-tests.sh]: I would also rather keep it the current way if there is only a one-second speedup...

**Comment by lzutao** [comment]
I would close this PR. 
I close it not because I feel that using archive is neglectable 1st faster using git.
But there are regression in clippy run-time in integration tests that I cannot resolve.
I tried to resolve it in #4712 by:

* using git again and measure run-time
* using git again, remove .git dir and measure run-time

But none of them causes regressions in run-time like this PR.
Unless I could resolve that regression, I close this PR.

**Pull request closed without merging**